src/num/vecops.c

```diff
@@ -145,25 +145,39 @@ static double zl1norm(long N, const complex float* vec)
 	return res;
 }
 
+
+static void axpbz(long N, float* dst, const float a1, const float* src1, const float a2, const float* src2)
+{
+	if (0. != a1 && 0 != a2)
+		for (long i = 0; i < N; i++)
+			dst[i] = a1 * src1[i]  + a2 * src2[i];
+	else if (0. == a1 && 0 != a2)
+		for (long i = 0; i < N; i++)
+			dst[i] = a2 * src2[i];
+	else if (0. != a1 && 0 == a2)
+		for (long i = 0; i < N; i++)
+			dst[i] = a1 * src1[i];
+	else
+		for (long i = 0; i < N; i++)
+			dst[i] = 0.;
+}
```

Comment:
I would be more readable to just call it into 'smul' and 'clear' for the special cases (of course, 'smul' then cannot be implemented in terms of this function). The other (even better) alternative is to just remove all special cases and leave it to the compiler.

Reply:
Will the compiler know to do that? I based the special cases since a similar thing was done for axpy previously. I can get rid of the special cases.

Reply:
If the call comes from a function in the same file (translation unit) and it explicitly passes a zero which leads to simplifications, I would expect the compiler to automatically specialize the function. I know clang and gcc can do this, but it might not work in all cases. Anyway, the general recommendation is to optimize manually only after having good data from profiling that is actually required.